libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -1066,7 +1068,7 @@ void AC_WPNav::wp_speed_update(float dt)
 {
     // return if speed has not changed
     float curr_max_speed_xy_cms = _pos_control.get_max_speed_xy();
-    if (is_equal(_wp_desired_speed_xy_cms, curr_max_speed_xy_cms)) {
+    if (_iswpspd_done_recalc || is_equal(_wp_desired_speed_xy_cms, curr_max_speed_xy_cms)) {
```

Comment:
I don't see how this is less computationally expensive.  You are still requesting it perform is_equal every loop.  Unless I'm missing something?
I think you could remove the is_equal statement and just have the if statement evaluate _iswpspd_done_recalc.  

@rmackay9  this is your call as I'm not a computer programmer by trade.  So I don't necessarily look for the most efficient ways to code.

Reply:
The "||" oprator would ban the second syntax to take effect.
Say if flag __iswpspd_done_recalc  is true the condition would directly satisfy and break then.

Reply:
I would like to commit a better one to this if you're not that pleased.